scripts/lib/basic_youtube_upload.py

```diff
@@ -130,20 +151,84 @@ def resumable_upload(request):
             error = 'A retriable error occurred: %s' % e
   
         if error is not None:
-            print(error)
+            debug(error)
             retry += 1
             if retry > MAX_RETRIES:
                 raise ValueError(error)
     
             max_sleep = 2 ** retry
             sleep_seconds = random.random() * max_sleep
-            print('Sleeping %f seconds and then retrying...' % sleep_seconds)
+            debug('Sleeping %f seconds and then retrying...' % sleep_seconds)
             time.sleep(sleep_seconds)
 
+def get_known_playlist(title):
+    return PLAYLIST_IDS[title]
+
+def get_playlist(youtube, title):
+    """Return users's playlist ID by title (None if not found)"""
+    playlists = youtube.playlists()
+    request = playlists.list(mine=True, part="id,snippet")
+    current_encoding = locale.getpreferredencoding()
+    
+    while request:
+        results = request.execute()
+        for item in results["items"]:
+            t = item.get("snippet", {}).get("title")
+            existing_playlist_title = (t.encode(current_encoding) if hasattr(t, 'decode') else t)
+            if existing_playlist_title == title:
+                return item.get("id")
+        request = playlists.list_next(request, results)
+
+def create_playlist(youtube, title, privacy):
+    """Create a playlist by title and return its ID"""
+    debug("Creating playlist: {0}".format(title))
```

Comment:
Better to use an f-string for this these days:

```suggestion
    debug(f"Creating playlist: {title}")
```

(You could probably do this all the places `%` interpolation is used, too, but ¯\\\_(ツ)\_/¯)